backend/equipmentposts/views.py

```diff
@@ -0,0 +1,163 @@
+from rest_framework.response import Response
+from rest_framework import viewsets
+from rest_framework import permissions
+from rest_framework.decorators import action
+from rest_framework.pagination import PageNumberPagination
+from django.db.models import Q, F, Func, IntegerField
+from datetime import datetime
+# Create your views here.
+
+class EquipmentPostsPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+class EquipmentViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    authentication_classes = [JWTAuthentication]
+    pagination_class = EquipmentPostsPagination
+    queryset = EquipmentPost.objects.all()
+    serializer_class = EquipmentSerializer
+    JWTauth = JWTAuthentication()
+    lookup_field = "id"
+
+    def wrap(self, request, data):
+        queryset = User.objects
+        owner = queryset.filter(id=data["owner"]).get().username
+        response = \
+            {
+                "@context": "https://www.w3.org/ns/activitystreams",
+                "summary": owner + " posted an equipment",
+                "type": "Create",
+                "actor":
+                    {
+                    "type": "Person",
+                    "name": owner
+                    },
+                "object":
+                    {
+                        "type": "Equipment",
+                        "postId": data["id"],
+                        "ownerId": data["owner"],
+                        "content": data["content"],
+                        "title": data["title"],
+                        "creationDate": data["creation_date"],
+                        "numberOfClicks": 0,
+                        "location":
+                            {
+                            "name": data["location"],
+                            "type": "Place",
+                            "longitude": data["longitude"],
+                            "latitude": data["latitude"],
+                            "units": "m"
+                            },
+                        "url": data["url"],
+                        "sport": data["sport"],
+                        "equipmentMinSkillLevel": data[
+                            "min_skill_level"],
+                        "equipmentMaxSkillLevel": data[
+                            "max_skill_level"],
+                        "equipmentType": data["equipment_type"]
+                    }
+            }
+
+        return response
+
+    def authenticate(self):
+        user, _ = self.JWTauth.authenticate(self.request)
+        return user.id == int(self.request.data["owner"])
+
+    def get_queryset(self):
+        queryset = EquipmentPost.objects.all()
+
+        # get all parameters for search
+        query = self.request.query_params.get('query')
+
+        location = self.request.query_params.get('location')
+
+        min_date = self.request.query_params.get('min_date')
```

Comment:
```suggestion

```
Equipment doesn't have a date field, right? This part must be left from eventposts code.

Reply:
Yeah sorry, I thought this was for filtering by the creation date. I will remove them.